api.go

```diff
@@ -25,5 +26,12 @@ func uploadFile(db *gorm.DB, p providers.StorageProvider) func(w http.ResponseWr
 		if err := p.UploadFile(data, name); err != nil {
 			http.Error(w, err.Error(), http.StatusBadRequest)
 		}
+		var tmp []byte
+		data.Read(tmp)
+		data.Close()
+		size := uint(len(tmp))
```

Comment:
Let’s use the “Content-Length” header for now (panic before uploading the file if it’s not provided). The thing is that we need to support large files and storing it in memory would be expensive.